changelog-extractor.py

```diff
@@ -125,19 +125,42 @@ def send_message_to_slack(text, channel, token, username) :
 def main():
     new_commit = os.environ['CI_COMMIT_SHA']
     old_commit = ""
-    if "CI_BUILD_TAG" in os.environ:
-        print 'there is a tag'
+
+    there_is_a_tag = "CI_BUILD_TAG" in os.environ 
+    there_is_a_release_tag = there_is_a_tag and os.environ["CI_BUILD_TAG"].endswith("-rc")
+    there_is_a_sos_tag = there_is_a_tag and os.environ["CI_BUILD_TAG"].endswith("-sos")
+    
+    # development branch
+    if not there_is_a_tag:
+        print "development case"
+        old_tag = run_command("git describe --abbrev=0 --match v*-rc ")
+   
+    # release case
+    if there_is_a_release_tag:
+        print "release case"
+        new_tag = os.environ["CI_BUILD_TAG"]
+        old_tag = run_command("git describe --abbrev=0 --tags "+ new_tag +"^ --match v*[0-9]-rc")
```

Comment:
Is this regex right?
This is the one used by other scripts:
v([\d]+\.){2}([\d]+)(-rc)

Your regex seems to work for v11-rc but not for v11.11.11-rc which is the format we are using

Reply:
it actually works for v11.11.11-rc

try to execute command 
git describe --abbrev=0 --match v*[0-9]-rc
git describe --abbrev=0 --match v*[0-9]
git describe --abbrev=0 --match v*[0-9]-sos

Reply:
In order to test "release case" 
1) add a tag with -rc suffix
2) run command git describe --abbrev=0 --tags v6.40.1-rc^ --match v*[0-9]-rc

Reply:
I was trying to use expression from other scripts, but it looks like we need to escape some characters in v([\d]+.){2}([\d]+)(-rc) in order to make it work in --match